From fe7613a3b9f2443cc11917826348d4521f267c96 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 29 Jun 2018 15:48:33 +0800
Subject: [PATCH] testsuite ignore elflint

On many architectures this test fails because binaries/libs produced by
binutils don't pass elflint. However elfutils shouldn't FTBFS because of this.

So we run the tests on all archs to see what breaks, but if it breaks we ignore
the result (exitcode 77 means: this test was skipped).

Upstream-Status: Pending [from debian]
testsuite-ignore-elflint.diff and rebase to 0.172

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 tests/run-elflint-self.sh |  2 +-
 tests/test-subr.sh        | 15 +++++++++++++++
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/tests/run-elflint-self.sh b/tests/run-elflint-self.sh
index 58fa7d0..85d21a5 100755
--- a/tests/run-elflint-self.sh
+++ b/tests/run-elflint-self.sh
@@ -18,5 +18,5 @@
 
 . $srcdir/test-subr.sh
 
-testrun_on_self ${abs_top_builddir}/src/elflint --quiet --gnu-ld
+testrun_on_self_skip ${abs_top_builddir}/src/elflint --quiet --gnu-ld
 testrun_on_self_compressed ${abs_top_builddir}/src/elflint --quiet --gnu-ld
diff --git a/tests/test-subr.sh b/tests/test-subr.sh
index 09f428d..26f61f1 100644
--- a/tests/test-subr.sh
+++ b/tests/test-subr.sh
@@ -201,3 +201,18 @@ testrun_on_self_quiet()
   # Only exit if something failed
   if test $exit_status != 0; then exit $exit_status; fi
 }
+
+# Same as testrun_on_self(), but skip on failure.
+testrun_on_self_skip()
+{
+  exit_status=0
+
+  for file in $self_test_files; do
+      testrun $* $file \
+	  || { echo "*** failure in $* $file"; exit_status=77; }
+  done
+
+  # Only exit if something failed
+  if test $exit_status != 0; then exit $exit_status; fi
+}
+
-- 
2.7.4

